### PR TITLE
fix: add keyboard handlers to sp-table-cell-head

### PR DESCRIPTION
**Pull request by jnurthen**
## Description

Adds keyboard handlers to sp-table-cell-head
## Related issue(s)

#4472

## Motivation and context

Keyboard handlers needed for accessibility.

## How has this been tested?

-   [x] _Test case 1_
    1. Go to [Storybook](https://jnurthen-issue4472--spectrum-web-components.netlify.app/storybook/)
    2. Tab to Column Title (sortable)
    3. Press Enter
    4. sort activates
    5. Press and release space
    6. sort activates

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/table/src/TableHeadCell.ts [hunk 1]
```diff
@@ -58,6 +58,45 @@ export class TableHeadCell extends SpectrumElement {
...
+        switch (code) {
+            case 'Space':
+                this.removeEventListener('keyup', this.handleKeyup);
+                this.click();
+                break;
+            /* c8 ignore next 2 */
+            default:
+                break;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (jnurthen) [reply packages/table/src/TableHeadCell.ts:98]: honestly I copied this from swatch which was doing the same thing. Happy to simplify if that makes sense

File: packages/table/src/TableHeadCell.ts [hunk 2]
```diff
@@ -58,6 +58,45 @@ export class TableHeadCell extends SpectrumElement {
...
+        switch (code) {
+            case 'Space':
+                event.preventDefault();
+                this.addEventListener('keyup', this.handleKeyup);
+                break;
+            /* c8 ignore next 2 */
+            default:
+                break;
+        }
+    }
```
Comment (Westbrook) [inline packages/table/src/TableHeadCell.ts:72]: Keydown, in previous consumptions, in this way was meant to allow for `this.active = true` while the `Space` key was doing to better mimic the interaction you get with a `<button>` element. This element currently includes styles for this interaction, but not the property to power those styles. We probably should...

Can you add this to the beginning of the class:
```
@property({ type: Boolean, reflect: true })
public active = false;
```

`this.active = true;` here. And `this.active = false;` to the `handleKeyup()` logic? Then we should be good to go here.

File: packages/table/src/TableHeadCell.ts [hunk 3]
```diff
@@ -58,6 +61,47 @@ export class TableHeadCell extends SpectrumElement {
...
+            case 'NumpadEnter':
+                this.click();
+                break;
+            /* c8 ignore next 2 */
+            default:
+                break;
+        }
+    }
+
+    protected handleKeyup(event: KeyboardEvent): void {
```
Comment (blunteshwar) [inline packages/table/src/TableHeadCell.ts:91]: Why is 'space' handled in `handleKeyup` and `handleKeydown` and not directly in `handleKeypress` like `enter`

Comment (jnurthen) [reply packages/table/src/TableHeadCell.ts:91]: Space and enter work slightly differently on native buttons.
Enter activates on the press event, space activates on the up event. This is intended to mimic that. A user should be able to press and hold space on a button, tab away from the button and then release space without activating it.

Comment (blunteshwar) [reply packages/table/src/TableHeadCell.ts:91]: Understood!

**Review by blunteshwar — APPROVED** [review]

File: packages/table/src/TableHeadCell.ts (hunk 2 above)
Comment (Rajdeepc) [reply packages/table/src/TableHeadCell.ts]: @jnurthen Can you please make sure this is added as requested above!

**Review by Rajdeepc — CHANGES_REQUESTED** [review]
One small addition needs to go in to the definition of this class. Can you make sure it is added before we can land this!

File: packages/table/src/TableHeadCell.ts (hunk 2 above)
Comment (jnurthen) [reply packages/table/src/TableHeadCell.ts]: I think I did in https://github.com/adobe/spectrum-web-components/pull/4473/commits/550b46a5320f5baf36887dd2e9e6957a5ef6f062

**Review by blunteshwar — APPROVED** [review]

**Review by Rajdeepc — APPROVED** [review]
LGTM!

**Pull request merged**